### PR TITLE
Xml2csv

### PR DESCRIPTION
**Pull request by ashishpriyadarshiCIC**
Co-authored by @coolalexzb 
Further working on the work done by @coolalexzb  for GSoC 2020

### CODE REVIEW

File: retriever/lib/engine_tools.py [hunk 1]
```diff
@@ -273,3 +274,51 @@ def set_proxy():
...
                     os.environ[i] = os.environ[proxy]
                 break
+
+
+def xml2dict(dic, node, level):
+    """Convert xml to dict type.
+
+    """
+    if not node.tag:
+        print(type(node))
```
Comment (apoorvaeternity) [inline retriever/lib/engine_tools.py:284]: Remove this print statement.

File: retriever/lib/engine_tools.py [hunk 2]
```diff
@@ -273,3 +274,51 @@ def set_proxy():
...
+        return
+    vals = dict()
+    for child in node:
+        key = child.tag.strip() + '_' + str(level)
```
Comment (apoorvaeternity) [inline retriever/lib/engine_tools.py:288]: Use python `f-strings`.

File: retriever/lib/engine_tools.py [hunk 3]
```diff
@@ -273,3 +274,51 @@ def set_proxy():
...
+        if key not in dic:
+            dic[key] = []
+        if child.attrib:
+            if key not in vals:
+                vals[key] = [child.attrib]
+            else:
+                vals[key].append(child.attrib)
+
+        if child.text and child.text.strip():
```
Comment (apoorvaeternity) [inline retriever/lib/engine_tools.py:297]: Can we use `if child.text.strip():` only?

File: retriever/lib/engine_tools.py [hunk 4]
```diff
@@ -273,3 +274,51 @@ def set_proxy():
...
+            if key not in vals:
+                vals[key] = [child.text]
+            else:
+                vals[key].append(child.text)
+        if child:
+            xml2dict(dic, child, level + 1)
+
+    for k in vals:
+        if len(vals.keys()) == 1:
```
Comment (apoorvaeternity) [inline retriever/lib/engine_tools.py:306]: `if len(vals) == 1`

**Comment by henrykironde** [comment]
ref Ref: #1506

**Pull request closed without merging**